src/java/org/apache/cassandra/db/commitlog/CommitLogReplayer.java

```diff
@@ -53,7 +53,15 @@
 import org.apache.cassandra.schema.TableMetadataRef;
 import org.apache.cassandra.utils.FBUtilities;
 import org.apache.cassandra.utils.WrappedRunnable;
+import org.jctools.maps.NonBlockingHashMap;
 
+/**
+ * Replays commit logs (reads commit logs and flushes new sstables).
+ *
+ * Note that instances of this class are meant to be used for a single replay only. Do not reuse the same
```

Comment:
let's add an assertion that multiple replay doesn't happen (if there isn't one already).

Reply:
good idea